### ISSUE TITLE
Don't have password to the database saved in the source code

### ISSUE THREAD

**Issue by C0DK** [Trivivium/devops-2k20#95]
Not sure why we have this in the first place. I assumed we all had security 1 :upside_down_face: 

anywho we should use some environmental constant or something.

**Comment by Trivivium** [comment]
I like the idea of environment variables.

If I remember correctly the `IConfiguration` instance available in the `Startup.cs` class by default reads environment variables. These should take precedence over all other configurations with the exception of command line arguments. 

However, we can also look into something called UserSecrets. Normally is these used only during development as the actual values are stored in plaintext on the host, but it would move the sensitive information out of the repository. However, I'm unsure if we can enable these in a production environment whereas you would normally used something like Azure Key Vault etc.

https://docs.microsoft.com/en-us/aspnet/core/security/app-secrets?view=aspnetcore-3.1&tabs=windows

**Comment by Trivivium** [comment]
We can also look into Docker secrets through Docker Compose... The main docker secrets solution requires us to run the application in Docker Swarm, but it seems Compose can "fake" it.

I think this is the way to go as the solution I proposed above where using user secrets in production isn't supported and we would have to "hack" around it.

https://docs.docker.com/engine/swarm/secrets/
https://blog.mikesir87.io/2017/05/using-docker-secrets-during-development/

**Comment by andersbjergfelt** [comment]
That looks good. I just stumbled upon this one:

https://docs.docker.com/compose/environment-variables/

TLDR; 
environment:
    - PW

docker-compose run -e PW=thispassrocks

**Comment by C0DK** [comment]
@andersbjergfelt that solution looks great. I don't think we should put too much effort into it. Worth noting is simply issues we've had so we can reflect on them at our exams. This is a big devops problem we learned. And all the different options are all great. Maybe we could add them to choices.md? :) both of the options are worth discussing.

**Comment by andersbjergfelt** [comment]
I'll make a PR. 

I agree with adding them to choices.md and use them to reflect/discuss

**Comment by andersbjergfelt** [comment]
It is more than just doing docker env. It is also shown in source code (Start.cs) I'll try to find a way to hide it

**Comment by C0DK** [comment]
merged with #96
